warehouse/internal/model/upload.go

```diff
@@ -82,6 +94,13 @@ type UploadJob struct {
 	LoadFileGenStartTime time.Time
 }
 
+type JobErrorType string
+
+type JobError struct {
+	Type   JobErrorType
+	Format *regexp.Regexp
```

Comment:
I am thinking of the following approach to allow the format to be described using both regex, and other ways as well.

```go
type Matcher interface {
     MatchString(string) bool
}

type Contains string

func (c Contains) MatchString(substr string) bool {
      return strings.Contains(string(c), substr)
}
```

```suggestion
	Format Matcher
```

Reply:
+1 for the matcher interface. We would abstract the underlying pattern matching and then test in whatever fashion we like.